odl/space/fspace.py

```diff
@@ -359,192 +481,416 @@ def element(self, fcall=None, fapply=None):
             If fcall is a `FunctionSetVector`, it is wrapped
             as a new `FunctionSpaceVector`.
 
-        fapply : `callable`, optional
-            The actual instruction for in-place evaluation.
-            Its first argument must be the `FunctionSet.range` element
-            or the array of such (vectorization) to which the
-            result is written.
-
-            If fapply is a `FunctionSetVector`, it is wrapped
-            as a new `FunctionSpaceVector`.
+        vectorized : bool
+            Whether ``fcall`` supports vectorized evaluation.
 
         Returns
         -------
         element : `FunctionSpaceVector`
-            The new element.
+            The new element, always supports vectorization
+
+        Notes
+        -----
+        If you specify ``vectorized=False``, the function is decorated
+        with a vectorizer, which makes two elements created this way
+        from the same function being regarded as *not equal*.
         """
-        if fcall is None and fapply is None:
+        if fcall is None:
             return self.zero()
         else:
-            return super().element(fcall, fapply)
+            if not callable(fcall):
+                raise TypeError('function {!r} is not callable.'.format(fcall))
+            if not vectorized:
+                if self.field == RealNumbers():
```

Comment:
That sounds like a good idea. Actually we could do that for some more sets, e.g. `Integers` and `Strings`. So at least in these cases we could have safe choices for data types and wouldn't have to guess wildly.

And besides that, we could even provide a full list of data types so we can check `if dtype in myset.dtypes:`